src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/CngKey.EC.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.Cryptography;
+using System.Diagnostics;
+
+namespace System.Security.Cryptography
+{
+    public sealed partial class CngKey : IDisposable
+    {
+        /// <summary>
+        /// Does the key use elliptic curve cryptography
+        /// </summary>
+        /// <returns></returns>
+        internal bool IsEcc()
+        {
+            return (this.AlgorithmGroup == CngAlgorithmGroup.ECDiffieHellman ||
```

Comment:
Style rules say to only use `this.` if removing it would change the execution of the program.